client/components/domains/use-my-domain/index.jsx

```diff
@@ -0,0 +1,175 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { useCallback, useEffect, useState, useRef } from 'react';
+import { __, sprintf } from '@wordpress/i18n';
+import { connect } from 'react-redux';
+import { Card, Button } from '@automattic/components';
+import { BackButton } from '@automattic/onboarding';
+import { createElement, createInterpolateElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import Gridicon from 'calypso/components/gridicon';
+import FormattedHeader from 'calypso/components/formatted-header';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormInputValidation from 'calypso/components/forms/form-input-validation';
+import FormButton from 'calypso/components/forms/form-button';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import page from 'page';
+
+/**
+ * Style dependencies
+ */
+
+import './style.scss';
+/**
+ * Image dependencies
+ */
+import domainIllustration from 'calypso/assets/images/illustrations/domain.svg';
+import { domainUseYourDomain } from 'calypso/my-sites/domains/paths';
+
+function UseMyDomain( { goBack, initialQuery, selectedSite } ) {
+	const [ domainName, setDomainName ] = useState( initialQuery ?? '' );
+	const [ domainNameValidationError, setDomainNameValidationError ] = useState();
+	const domainNameInput = useRef( null );
+	const initialValidation = useRef( null );
+
+	const baseClassName = 'use-my-domain';
+
+	const illustration = domainIllustration && (
+		<img src={ domainIllustration } alt="" width={ 160 } />
+	);
+
+	const validateDomainName = useCallback( () => {
+		if ( ! domainName ) {
+			setDomainNameValidationError( __( 'Please enter your domain before continuing.' ) );
+			return false;
+		}
+
+		if ( ! /^[a-zA-Z0-9][a-zA-Z0-9-]{1,61}[a-zA-Z0-9](?:\.[a-zA-Z]{2,})+$/.test( domainName ) ) {
```

Comment:
Is this regex used anywhere else? Maybe it would be nice to move it as a `isValidDomainName` function in a utility file? Just an idea, not sure if it makes sense.

Reply:
It's not used anywhere else that I know of.  Usually we do this sort of validation on the back end, but in this case I didn't think it was worth the round trip since we'll do a more thorough check on the back end in the next step anyway.

I'm a bit hesitant to move this into it's own utility function since it's not really an exhaustive check, but I felt that allowing a simplified check like this was good enough to let us move to the next step.  I guess if we end up having multiple use cases for this sort of thing, then we can create a utility function for it at that point.

Reply:
Cool cool, that makes sense 👍